### PR TITLE
fix: Define both 'gflags' and 'gflags::gflags' alias targets

### PR DESCRIPTION
**Pull request by schuhschuh**
When `gflags` library is included directly as subproject, define both `gflags` and `gflags::gflags` alias targets. Note that other library targets have no `gflags::` package namespace prefix in this case either.

Extends #247 to maintain backwards compatibility.

### CODE REVIEW

**Comment by schuhschuh** [comment]
@Mizux This merge request exports the targets once with and once without namespace. By default, the targets without `gflags::` namespace prefix are imported by `gflags-config.cmake`, i.e., `find_package(gflags)` in order to maintain backwards compatibility. You can use the namespaced import targets by setting `set(GFLAGS_USE_TARGET_NAMESPACE TRUE)` or setting it via `-D` switch when running CMake. Is that a viable solution for your use case?

File: cmake/config.cmake.in [hunk 1]
```diff
@@ -7,7 +7,21 @@ set (@PACKAGE_PREFIX@_VERSION_MINOR  @PACKAGE_VERSION_MINOR@)
 set (@PACKAGE_PREFIX@_VERSION_PATCH  @PACKAGE_VERSION_PATCH@)
 
 # import targets
-include ("${CMAKE_CURRENT_LIST_DIR}/@EXPORT_NAME@.cmake")
+if (NOT DEFINED @PACKAGE_PREFIX@_USE_TARGET_NAMESPACE)
+  set (@PACKAGE_PREFIX@_USE_TARGET_NAMESPACE FALSE)
```
Comment (schuhschuh) [inline cmake/config.cmake.in]: This default value may be changed to `TRUE` in later releases.

File: cmake/config.cmake.in [hunk 2]
```diff
@@ -7,7 +7,21 @@ set (@PACKAGE_PREFIX@_VERSION_MINOR  @PACKAGE_VERSION_MINOR@)
...
+endif ()
+if (@PACKAGE_PREFIX@_USE_TARGET_NAMESPACE)
+  include ("${CMAKE_CURRENT_LIST_DIR}/@EXPORT_NAME@.cmake")
+  set (@PACKAGE_PREFIX@_TARGET_NAMESPACE @PACKAGE_NAME@)
```
Comment (Mizux) [inline cmake/config.cmake.in]: @PACKAGE_NAME@::

Comment (schuhschuh) [reply cmake/config.cmake.in]: The `::` is appended in the following if block, cf. `@PACKAGE_PREFIX@_TARGET_PREFIX`.

**Comment by Mizux** [comment]
I guess, It should do the tricks.
Thanks !

**Comment by schuhschuh** [comment]
Tested using the gflags-example project (and update the latter to work with either `GFLAGS_USE_TARGET_NAMESPACE` setting).

**Pull request merged**